controllers/cnbi/controller.go

```diff
@@ -303,37 +317,54 @@ func (r *CustomNBImageReconciler) ReconcilePipelineRun(name string, ctx *context
 			}
 			controllerutil.SetControllerReference(r.CNBi, pipelineRun, r.Scheme)
 
-			if err := r.Create(*ctx, pipelineRun); err != nil {
+			if err := r.Create(ctx, pipelineRun); err != nil {
 				logger.Error(err, "Unable to create PipelineRun")
-				updateStatus(metav1.ConditionTrue, "ErrorPipelineRunCreate", fmt.Sprintf("Unable to create PipelineRun. %s", err))
-				return err
+
+				setCondition(newStatus, meteorv1alpha1.ErrorPipelineRunCreate, metav1.ConditionTrue, "PipelineRunCreateFailed", err.Error())
+				return newStatus
 			}
-			updateStatus(metav1.ConditionTrue, "SuccessPipelineRunCreate", "Tekton PipelineRun was created.")
-			return nil
+			setCondition(newStatus, meteorv1alpha1.PipelineRunCreated, metav1.ConditionTrue, "PipelineRunCreated", fmt.Sprintf("%s PipelineRun created successfully", name))
+			return newStatus
 		}
-		logger.Error(err, "Error fetching PipelineRun")
 
-		updateStatus(metav1.ConditionFalse, "ErrorPipelineRun", fmt.Sprintf("Reconcile resulted in error. %s", err))
-		return err
+		logger.Error(err, "Error fetching PipelineRun")
+		setCondition(newStatus, meteorv1alpha1.GenericPipelineError, metav1.ConditionTrue, "PipelineRunGenericError", err.Error())
+		return newStatus
 	}
 
 	if len(pipelineRun.Status.Conditions) > 0 {
-		if len(pipelineRun.Status.Conditions) != 1 {
+		if len(pipelineRun.Status.Conditions) != 1 { // TODO observe tekton project if they stay with just one condition all the time
 			logger.Error(nil, "Tekton reported multiple conditions")
 		}
-		condition := pipelineRun.Status.Conditions[0]
-		updateStatus(metav1.ConditionStatus(condition.Status), condition.Reason, condition.Message)
+
+		if pipelineRun.Labels["cnbi.thoth-station.ninja/pipeline"] == "import" {
+			if pipelineRun.Status.Conditions[0].Status == v1.ConditionFalse && pipelineRun.Status.Conditions[0].Type == "Succeeded" {
+				setCondition(newStatus, meteorv1alpha1.ImageImportReady, metav1.ConditionFalse, "ImageImportNotReady", "Import failed, this could be due to the repository to import from does not exist or is not accessible")
+				setCondition(newStatus, meteorv1alpha1.PipelineRunCompleted, metav1.ConditionTrue, "PipelineRunCompleted", "The PipelineRun has been completed, but the Image could not be imported!")
+				removeCondition(newStatus, meteorv1alpha1.PipelineRunCreated)
+			} else if pipelineRun.Status.Conditions[0].Status == v1.ConditionTrue && pipelineRun.Status.Conditions[0].Type == "Succeeded" {
+				setCondition(newStatus, meteorv1alpha1.ImageImportReady, metav1.ConditionTrue, "ImageImportReady", "Import succeeded, the image is ready to be used")
+				setCondition(newStatus, meteorv1alpha1.PipelineRunCompleted, metav1.ConditionTrue, "PipelineRunCompleted", "The PipelineRun has been completed, Image is importend")
+				removeCondition(newStatus, meteorv1alpha1.PipelineRunCreated)
```

Comment:
Should we not check directly the ImageStream to see if our image is ready and available ?

Reply:
let's review if storing the state in the ImageStream is really a good solution.